include/turtle/object.hpp

```diff
@@ -51,7 +51,7 @@ namespace detail
             : impl_( boost::make_shared< detail::object_impl >() )
         {}
     protected:
-        ~object()
+        virtual ~object()
```

Comment:
Can you explain why this change?

Reply:
Oh yeah, this accidently made it to here. It is the issue from https://github.com/mat007/turtle/pull/44 - that PR came from a former colleague. Unfortunately I did not have the time to step deep into the topic, so I just made the ctor virtual to get our socket mock to work again 🤷‍♂